Diagnostics/FreeBusyChecker/Functions/OnPremDAuthFunctions.ps1

```diff
@@ -0,0 +1,631 @@
+            $serWSSecurityAuthenticationColor = "Green"
+        }
+    } else {
+        Write-Host -ForegroundColor Red " WSSecurityAuthentication is NOT correct."
+        foreach ( $ser in $Script:AutoDiscoveryVirtualDirectory) {
+            Write-Host " $($ser.Identity)"
+            Write-Host -ForegroundColor Red "  WSSecurityAuthentication: $($ser.WSSecurityAuthentication)"
+            $serWSSecurityAuthenticationColor = "Red"
+            Write-Host " $($ser.Identity) "
+            $AutoD_VD_Identity = $ser.Identity
+            $AutoD_VD_Name = $ser.Name
+            $AutoD_VD_InternalAuthenticationMethods = $ser.InternalAuthenticationMethods
+            $AutoD_VD_ExternalAuthenticationMethods = $ser.ExternalAuthenticationMethods
+            $AutoD_VD_WSAuthentication = $ser.WSSecurityAuthentication
+            $AutoD_VD_WSAuthenticationColor = "green"
+            $AutoD_VD_WindowsAuthentication = $ser.WindowsAuthentication
+            if ($AutoD_VD_WindowsAuthentication -eq "True") {
+                $AutoD_VD_WindowsAuthenticationColor = "green"
+            } else {
+                $AutoD_VD_WindowsAuthenticationColor = "red"
+            }
+            $AutoD_VD_InternalNblBypassUrl = $ser.InternalNblBypassUrl
+            $AutoD_VD_InternalUrl = $ser.InternalUrl
+            $AutoD_VD_ExternalUrl = $ser.ExternalUrl
+            autoDVDHtmlNotOK
+            Write-Host -ForegroundColor Green "  WSSecurityAuthentication: $($ser.WSSecurityAuthentication)"
+            $serWSSecurityAuthenticationColor = "Red"
+        }
+        Write-Host -ForegroundColor White "  Should be True "
+    }
+    Write-Host -ForegroundColor White "`n  WindowsAuthentication:"
+    if ($Script:AutoDiscoveryVirtualDirectory.WindowsAuthentication -eq "True") {
+        foreach ( $ser in $Script:AutoDiscoveryVirtualDirectory) {
+            Write-Host " $($ser.Identity) "
+            Write-Host -ForegroundColor Green "  WindowsAuthentication: $($ser.WindowsAuthentication)"
+        }
+    } else {
+        Write-Host -ForegroundColor Red " WindowsAuthentication is NOT correct."
+        foreach ( $ser in $Script:AutoDiscoveryVirtualDirectory) {
+            Write-Host " $($ser.Identity)"
+            Write-Host -ForegroundColor Red "  WindowsAuthentication: $($ser.WindowsAuthentication)"
+        }
+        Write-Host -ForegroundColor White "  Should be True "
+    }
+    Write-Host -ForegroundColor Yellow "`n  Reference: https://learn.microsoft.com/en-us/powershell/module/exchange/Get-AutoDiscovervirtualdirectory?view=exchange-ps"
+}
+function EWSVirtualDirectoryCheck {
+    Write-Host -ForegroundColor Green " Get-WebServicesVirtualDirectory -Server $($server)| Select Identity,Name,ExchangeVersion,*Authentication*,*url"
+    PrintDynamicWidthLine
+    $Script:WebServicesVirtualDirectory = Get-WebServicesVirtualDirectory -Server $server | Select-Object Identity, Name, ExchangeVersion, *Authentication*, *url -ErrorAction SilentlyContinue

```

Comment:
This is the second time running this cmdlet, we should avoid doing that.

Reply:
Corrected,  Get-WebServicesVirtualDirectory is only exectuted if not execiuted before. If it has been previuosly exevuted it now uses variable content. Correction will be available when changes are pushed.

Reply:
A new file was added to hold functions that are common to OAuth and Dauth

![image](https://github.com/user-attachments/assets/cfe6025f-fe75-475b-a6ec-4c7b7d667aa7)